static/js/apps/nl_interface/app.tsx

```diff
@@ -28,6 +29,38 @@ export function App(): JSX.Element {
   const [queries, setQueries] = useState<string[]>([]);
   const [contextList, setContextList] = useState<any[]>([]);
 
+  // Executes a single query through the search box.
+  function executePrompt(prompt: string) {
+    if (prompt) {
+      (
+        document.getElementById("query-search-input") as HTMLInputElement
+      ).value = prompt;
+      (
+        document.getElementById("rich-search-button") as HTMLButtonElement
+      ).click();
+    }
+  }
+
+  useEffect(() => {
+    // Runs each prompt (';' separated) 2s apart.
+    const urlPrompts = getUrlToken("q");
+    if (urlPrompts) {
+      const prompts = urlPrompts.split(";");
+      executePrompt(prompts.shift());
+      if (prompts.length) {
```

Comment:
hm, I think this would run max 2 prompts, did you want a while loop instead?

Reply:
The `prompts` array gets gradually `shift`ed in the interval until the array is empty. I also wanted to execute the first immediately.